### PR TITLE
Buffs IEDs

### PR DESCRIPTION
**Pull request by Fox-McCloud**
IEDs are virtually worthless in their current form; theyr'e pretty much completely outclassed by a molatov cocktail (which is reusable to boot).

In any event, this buffs IEDs so they actually generate a small explosion again in addition to always having flames.

Slap crafting for these has been removed now that the crafting system is well cemented into people's minds (anddd it was pretty bad, even as far as slap crafting goes).

Code changes:
- Adds in TG's base proc for mutable appearances; mutable appearances are inherent to BYOND.

:cl: Fox McCloud
tweak: IEDs now have a blast radius again
tweak: Slap crafting of IEDs is gone; use personal crafting instead
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/legcuffs.dm [hunk 1]
```diff
@@ -52,23 +52,12 @@
...
-				desc = "A trap used to catch bears and other legged creatures. <span class='warning'>There is an IED hooked up to it.</span>"
-			else
-				to_chat(user, "<span class='danger'>You shouldn't be reading this message! Contact a coder or someone, something broke!</span>")
-				IED = null
-				return
+		user.drop_item()
+		I.forceMove(src)
+		message_admins("[key_name_admin(user)] has rigged a beartrap with an IED.")
+		log_game("[key_name(user)] has rigged a beartrap with an IED.")
+		to_chat(user, "<span class='notice'>You sneak the [IED] underneath the pressure plate and connect the trigger wire.</span>")
```
Comment (FlattestGuitar) [inline code/game/objects/items/weapons/legcuffs.dm]: You sneak the the IED underneath...

Comment (Fox-McCloud) [reply code/game/objects/items/weapons/legcuffs.dm]: Ponies. Ponies why.

**Comment by Shadeykins** [comment]
Will this buff explosive lances at all, or no?

Also I assume it's on the same scale as old-school IED's, meaning they're able to damage (but not destroy) station plating?

**Comment by HugoLuman** [comment]
Finally, they have been literally worthless for the longest time.

**Comment by imsxz** [comment]
This buffs the shit out of explosive lances. IED + spear is the biggest meme vs nukies/sleeping carp/etc on TG, 1 hit and the target is out cold unable to use adrenals or anything (and usually burning). That's more of an issue with light explosions though; they have a something% chance to knock people unconsious when hit.

**Comment by tigercat2000** [comment]
LGTM

File: code/datums/mutable_appearance.dm [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// Mutable appearances are an inbuilt byond datastructure. Read the documentation on them by hitting F1 in DM.
+// Basically use them instead of images for overlays/underlays and when changing an object's appearance if you're doing so with any regularity.
+// Unless you need the overlay/underlay to have a different direction than the base object. Then you have to use an image due to a bug.
+
+// Mutable appearances are children of images, just so you know.
+
+/mutable_appearance/New()
+	..()
+	plane = FLOAT_PLANE // No clue why this is 0 by default yet images are on FLOAT_PLANE
```
Comment (tigercat2000) [inline code/datums/mutable_appearance.dm]: Can we please not do this? It makes mutable_appearances more annoying to use for anything other than floating overlays. You have to manually fuck with the plane if you wanted to, say, use these in human icon code.

**Pull request merged**